src/test/java/StringManipulationTest.java

```diff
@@ -55,95 +66,140 @@ public void testRemoveNthCharacter2() {
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        String result = manipulatedstring.removeNthCharacter(2, false);
+        assertEquals("Hlo ol!", result);
     }
 
+
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        String result = manipulatedstring.removeNthCharacter(5, true);
+        assertEquals("Hell , Wo ld!", result);
     }
 
+
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.removeNthCharacter(0, false);
+        });
     }
 
+
+
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        String result = manipulatedstring.removeNthCharacter(3, false);
+        assertEquals("Helo Wrl!", result);
     }
 
+
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("world");
+        String result = manipulatedstring.removeNthCharacter(2, true);
+        assertEquals("w r d", result);
     }
 
+
     @Test
-    public void testGeSubStrings1() {
+    public void testGetSubStrings1() {
```

Comment:
You need to add unit tests for all the if statements in `getSubstrings()` in StringManipulation.java
ex:
```
if (startWord <= 0 || endWord <= 0 || startWord > endWord) {
    throw new IllegalArgumentException("Invalid startWord or endWord values");
}
```
```
if (string == null || string.isEmpty()) {
    throw new IndexOutOfBoundsException("String is empty");
}
```
```
String[] words = string.trim().split("\\s+");
if (endWord > words.length) {
    throw new IndexOutOfBoundsException("String has fewer words than endWord");
}
```